clients/web/src/views/widgets/GroupAdminsWidget.js

```diff
@@ -21,46 +21,43 @@ var GroupAdminsWidget = View.extend({
         'click .g-demote-moderator': function (e) {
             var li = $(e.currentTarget).parents('.g-group-admins>li');
             var userid = li.attr('userid');
-            var view = this;
 
             confirm({
                 text: 'Are you sure you want to demote <b>' +
                     _.escape(li.attr('username')) + '</b> to a moderator?',
                 escapedHtml: true,
-                confirmCallback: function () {
+                confirmCallback: () => {
                     var user = new UserModel({_id: userid});
-                    view.model.off('g:promoted').on('g:promoted', function () {
+                    this.model.off('g:promoted').on('g:promoted', () => {
                         this.trigger('g:moderatorAdded');
-                    }, view).promoteUser(user, AccessType.WRITE);
```

Comment:
That's a good point. I've revered the places that the `this` context was removed from an `.on` binding.

All of this `.off().on()` code should go away soon anyway, once the Promise infrastructure is improved.